savant/utils/image.py

```diff
@@ -0,0 +1,512 @@
+        :param point: point to paste (x, y)
+        """
+
+        frame_roi = self._gpu_image.colRange(point[0], point[0] + image.width).rowRange(
+            point[1], point[1] + image.height
+        )
+        image.gpu_mat.copyTo(stream=self._cuda_stream, dst=frame_roi)
+
+    def rotate(
+        self, angle: float, bbox: Optional[RBBox] = None
+    ) -> Tuple['GPUImage', RBBox]:
+        """Rotates image on angle. If.
+
+        :param angle: angle to rotate in degrees
+        :param bbox: bounding box on image
+        :return: rotated image
+        """
+
+        rotation_matrix, resolution = get_rotation_matrix(
+            self, angle, (bbox.x_center, bbox.y_center)
+        )
+
+        res = cv2.cuda.warpAffine(
+            src=self._gpu_image,
+            M=rotation_matrix,
+            dsize=resolution,
+            stream=self._cuda_stream,
+        )
+
+        if bbox is not None:
+            return GPUImage(image=res, cuda_stream=self._cuda_stream), RBBox(
+                x_center=resolution[0] / 2,
+                y_center=resolution[1] / 2,
+                width=bbox.width,
+                height=bbox.height,
+                angle=bbox.angle - angle,
+            )
+        return GPUImage(image=res, cuda_stream=self._cuda_stream), RBBox(
+            x_center=resolution[0] / 2,
+            y_center=resolution[1] / 2,
+            width=self.width,
+            height=self.height,
+            angle=angle,
+        )
+
+    def resize(
+        self,
+        resolution: Tuple[int, int],
+        method: str = 'fit',
+        interpolation: int = cv2.INTER_LINEAR,
+    ) -> 'GPUImage':
+        """Resizes image to resolution.
+
+        :param resolution: resolution to resize [width, height]
+        :param method: method to resize one of ['fit', 'scale']
+        :param interpolation: interpolation method.
+            Oone of [cv2.INTER_NEAREST, cv2.INTER_AREA, cv2.INTER_LINEAR,
+            cv2.INTER_CUBIC, cv2.INTER_LANCZOS4]
+        :return: resized image
+        """
+        new_resolution = tuple(map(int, resolution))
+        if method == 'scale':
+            scale_factor = min(resolution[0] / self.width, resolution[1] / self.height)
+            new_resolution = (
+                int(self.width * scale_factor),
+                int(self.height * scale_factor),
+            )
+        resized_image = cv2.cuda.resize(
+            src=self._gpu_image,
+            dsize=new_resolution,
+            interpolation=interpolation,
+            stream=self._cuda_stream,
+        )
+        res = cv2.cuda.GpuMat(size=resolution, type=self._gpu_image.type(), s=0)
+        start_col = (resolution[0] - new_resolution[0]) // 2
+        start_row = (resolution[1] - new_resolution[1]) // 2
+        res_roi = res.colRange(start_col, start_col + new_resolution[0]).rowRange(
+            start_row, start_row + new_resolution[1]
+        )
+        resized_image.copyTo(stream=self._cuda_stream, dst=res_roi)
```

Comment:
Is it possible to use a `cv2.cuda.resize` call with the destination GpuMat argument, and avoid extra call to `copyTo()`?

Reply:
Yes, added